### PR TITLE
Use partial for Ruby HTTP libraries

### PR DESCRIPTION
**Pull request by wing328**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

- bug fix: isFaraday
- use mustache partial for configuration.mustache

cc @cliffano (2017/07) @zlx (2017/09) @autopp (2019/02)

### CODE REVIEW

**Comment by wing328** [comment]
@dkliban please have a look when you've time.

**Comment by wing328** [comment]
Local tests (Ruby Faraday) look good:
```
Finished in 25.58 seconds (files took 17.09 seconds to load)
277 examples, 0 failures, 3 pending
```

File: modules/openapi-generator/src/main/resources/ruby-client/configuration.mustache [hunk 1]
```diff
@@ -125,11 +96,20 @@ module {{moduleName}}
...
       @client_side_validation = true
+      {{#isFaraday}}
+      @ssl_verify = true
+      @ssl_verify_mode = nil
+      @ssl_ca_file = nil
+      @ssl_client_cert = nil
+      @ssl_client_key = nil
+      {{/isFaraday}}
+      {{^isFaraday}}
       @verify_ssl = true
```
Comment (dkliban) [inline modules/openapi-generator/src/main/resources/ruby-client/configuration.mustache]: You should only include the TLS settings for Typhoeus when not isFaraday.

Comment (dkliban) [reply modules/openapi-generator/src/main/resources/ruby-client/configuration.mustache]: Nevemind. I misread.

**Pull request merged**